components/feature/tabs/src/main/java/mozilla/components/feature/tabs/TabsUseCases.kt

```diff
@@ -299,16 +303,32 @@ class TabsUseCases(
      * Use case for restoring tabs.
      */
     class RestoreUseCase(
-        private val sessionManager: SessionManager,
-        private val select: SelectTabUseCase
+        private val sessionManager: SessionManager
     ) {
         /**
          * Restores the given list of [RecoverableTab]s.
          */
         operator fun invoke(tabs: List<RecoverableTab>, selectTabId: String? = null) {
-            sessionManager.restore(tabs)
+            sessionManager.restore(tabs, selectTabId)
+        }
 
-            selectTabId?.let { id -> select(id) }
+        /**
+         * Restores the given [BrowsingSession].
+         */
+        operator fun invoke(browsingSession: BrowsingSession) {
+            invoke(browsingSession.tabs, browsingSession.selectedTabId)
+        }
+
+        /**
+         * Restores the browsing session from the given [SessionStorage].
+         */
+        suspend operator fun invoke(storage: SessionStorage) = withContext(Dispatchers.IO) {
+            val browsingSession = storage.restore()
+            if (browsingSession != null) {
+                withContext(Dispatchers.Main) {
+                    invoke(browsingSession)
+                }
```

Comment:
Nice! This will also close a ticket a filed a while ago: https://github.com/mozilla-mobile/android-components/issues/8600.


Can we also dispatch `RestoreCompleteAction` here which Fenix does itself right now?